kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/clusternetworkaddressconfigprovider/IndexedIntInterval.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.internal.clusternetworkaddressconfigprovider;
+
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * A non-empty integer interval. The index for each value is the 0-based position of the
+ * value in the interval from smallest to largest.
+ * @param startInclusive the inclusive start integer
+ * @param endExclusive the exclusive end integer
+ */
+record IndexedIntInterval(int startInclusive, int endExclusive) implements IndexedIntSet {
+
+    private static final String START_INCLUSIVE = "[";
+    private static final String START_EXCLUSIVE = "(";
+    private static final String END_INCLUSIVE = "]";
+    private static final String END_EXCLUSIVE = ")";
+    private static final String INTEGER_GROUP = "(\\d*)";
```

Comment:
```suggestion
    private static final String INTEGER_GROUP = "(\\d+)";
```

You are missing a test case for 
```
assertDoesNotMatchIntervalPattern("[,3]");
```